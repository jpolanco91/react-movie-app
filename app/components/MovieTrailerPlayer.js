import React, { Component } from 'react';
import PropTypes from 'react-proptypes';
import Axios from 'axios';

class MovieTrailerPlayer extends Component {
  state = {
    currentMovieTrailerUrl: '',
  };

  componentDidMount() {
    const { apiUrl } = this.props;
    const { match } = this.props;
    const fullTrailerUrl = `${apiUrl}${match.params.movieId}`;

    Axios.get(fullTrailerUrl).then(({ data = [] }) => {
      this.setState({ currentMovieTrailerUrl: data.trailer });
    });
  }

  render() {
    const { currentMovieTrailerUrl } = this.state;
    return (
      <div>
        <a href={currentMovieTrailerUrl}>
          <span>Link to play trailer</span>
        </a>
      </div>
    );
  }
}

MovieTrailerPlayer.propTypes = {
  apiUrl: PropTypes.string.isRequired,
  match: PropTypes.objectOf(PropTypes.object),
};

MovieTrailerPlayer.defaultProps = {
  match: {},
};

export default MovieTrailerPlayer;
