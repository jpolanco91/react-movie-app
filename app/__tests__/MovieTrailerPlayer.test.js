import React from 'react';
import { create } from 'react-test-renderer';
import MovieTrailerPlayer from '../components/MovieTrailerPlayer';

test('MovieTrailerPlayer snapshot test', () => {
  const matchParams = { params: { movieId: '1' }, isExact: true, path: '', url: '' };
  const c = create(<MovieTrailerPlayer apiUrl="http://myapi.com/api/" match={matchParams} />);
  expect(c.toJSON()).toMatchSnapshot();
});
