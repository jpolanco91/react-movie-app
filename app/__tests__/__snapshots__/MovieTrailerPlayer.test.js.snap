// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTrailerPlayer snapshot test 1`] = `
<section
  className="section is-medium"
>
  <div
    className="container"
  >
    <div
      className="card"
    >
      <div
        className="card-image"
      >
        <div
          className="videoWrapper"
        >
          <iframe
            allow="autoplay"
            allowFullScreen={true}
            frameBorder="0"
            src="https://www.youtube.com/embed/undefined"
            title="video"
            width="100%"
          />
        </div>
      </div>
      <div
        className="card-content"
      >
        <div
          className="media-content"
        >
          <p
            className="title is-4"
          >
            
          </p>
          <p
            className="subtitle is-6"
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
